isis/src/control/apps/cnetwinnow/cnetwinnow.cpp

```diff
@@ -0,0 +1,437 @@
+#include "cnetwinnow.h"
+
+#include <geos_c.h>
+#include <geos/algorithm/ConvexHull.h>
+#include <geos/geom/CoordinateSequence.h>
+#include <geos/geom/CoordinateArraySequence.h>
+#include <geos/geom/Envelope.h>
+#include <geos/geom/Geometry.h>
+#include <geos/geom/GeometryFactory.h>
+#include <geos/geom/Polygon.h>
+
+#include <QHash>
+#include <QString>
+#include <QList>
+#include <QVector>
+
+#include "ControlMeasure.h"
+#include "ControlPoint.h"
+#include "PvlGroup.h"
+#include "Histogram.h"
+#include "IException.h"
+#include "IString.h"
+#include "Progress.h"
+#include "SerialNumber.h"
+#include "SerialNumberList.h"
+#include "UserInterface.h"
+
+#include "stdio.h"
+#include "string.h"
+
+using namespace std;
+using namespace Isis;
+
+namespace Isis {
+
+  void cubeConvexHullAndMeasures(QString &serialNum,ControlNet &net, double &area, int &validMeasures,
+                                 QList <ControlMeasure *> *measToIgnor=NULL);
+
+  void cnetwinnow(UserInterface &ui) {
+    //build a histogram from the control net
+    Progress progress;
+
+    //read the ControlNet
+    ControlNet net(ui.GetFileName("CNET"), &progress);
+
+    cnetwinnow(net, progress, ui);
+  }
+
+  void cnetwinnow(ControlNet &net, Progress &progress, UserInterface &ui) {
```

Comment:
From our conversation, I think we want to add the FileList object as a parameter too for these control apps.

Reply:
I agree but the a fromlist is only used to get the serial numbers as a serialNumList so I think that makes more sense to pass in this particular case

Reply:
Another param question, is it necessary to pass a progress object? Is it because of the control net constructor? Any way to simply pass in the network object then associate the cnet object with the progress object after the fact?

Reply:
Looks like it defaults to zero in the ControlNet.h file so we can default it in the cnetwinnow call as well

Reply:
We can at least get around needing it if we just want to call cnetwinnow.